src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -484,19 +560,105 @@ def run_sessions_query(
                 # Create entry in default dict:
                 output_groups[GroupKey(session_status=status)]
 
+    result_groups: Sequence[SessionsQueryGroup] = [
+        # Convert group keys back to dictionaries:
+        {"by": group_key.to_output_dict(), **group}  # type: ignore
+        for group_key, group in output_groups.items()
+    ]
+    result_groups = _reorder_result_groups_according_to_preflight_query_results(
+        ordered_preflight_filters, query.raw_groupby, result_groups, default_group_gen_func
+    )
+
     return {
-        "groups": [
-            # Convert group keys back to dictionaries:
-            {"by": group_key.to_output_dict(), **group}  # type: ignore
-            for group_key, group in output_groups.items()
-        ],
+        "groups": result_groups,
         "start": isoformat_z(metrics_results["start"]),
         "end": isoformat_z(metrics_results["end"]),
         "intervals": [isoformat_z(ts) for ts in metrics_results["intervals"]],
         "query": query.query,
     }
 
 
+def _reorder_result_groups_according_to_preflight_query_results(
+    ordered_preflight_filters: Dict[GroupByFieldName, Sequence[str]],
+    groupby: GroupByFieldName,
+    result_groups: Sequence[SessionsQueryGroup],
+    default_group_gen_func: Callable[[], Group],
+) -> Sequence[SessionsQueryGroup]:
+    """
+    Function that If a preflight query was run, then we want to preserve the order of results
+    returned by the preflight query
+    We create a mapping between the group value to the result group, so we are able
+    to easily sort the resulting groups.
+    For example, if we are ordering by `-release.timestamp`, we might get from
+    postgres a list of results ['1B', '1A'], and results from metrics dataset
+    [
+        {
+            "by": {"release": "1A"},
+            "totals": {"sum(session)": 0},
+            "series": {"sum(session)": [0]},
+        },
+        {
+            "by": {"release": "1B"},
+            "totals": {"sum(session)": 10},
+            "series": {"sum(session)": [10]},
+        },
+    ]
+    Then we create a mapping from release value to the result group:
+    {
+        "1A": [
+            {
+                "by": {"release": "1A"},
+                "totals": {"sum(session)": 0},
+                "series": {"sum(session)": [0]},
+            },
+        ],
+        "1B": [
+            {
+                "by": {"release": "1B"},
+                "totals": {"sum(session)": 10},
+                "series": {"sum(session)": [10]},
+            },
+        ],
+    }
+    Then loop over the releases list sequentially, and rebuild the result_groups
+    array based on that order by appending to the list the values from that mapping
+    and accessing it through the key which is the group value
+    """
+    if len(ordered_preflight_filters) == 1:
```

Comment:
what if ordered_preflight_filters > 1? 🤔

Reply:
There is validation against that early on in the function
https://github.com/getsentry/sentry/pull/34699/files#diff-9768b861d30daef14443451ae5e87b0352a6229bef88dfc9722fd06bad13afe1R786

Reply:
gotcha.